frontend/lib/framework/error-classes-list.cpp

```diff
@@ -537,4 +538,189 @@ void ErrorProcDefExplicitAnonFormal::write(ErrorWriterBase& wr) const {
   wr.code(fn, {formal});
 }
 
+/* parser errors */
+
+void ErrorParsing::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto errorMessage = std::get<std::string>(info);
+  assert(errorMessage.back() == '.' &&
+         "expected a period at the end of ErrorParsing message");
+  wr.heading(kind_, type_, loc, errorMessage);
```

Comment:
Do we want to print code here, too?

Reply:
I was on the fence about this one out of concern that we might potentially be given a weird location where only its start has meaning, but I think if that were to happen it would mean we need to fix the call site generating that location. I'll add it and make sure it doesn't give us any weird outputs.

Reply:
These are not actually well covered by tests because they only show up in detailed mode, which we aren't testing anywhere except in the per-specialized-error-class tests in `test/parsing/errors`. I think it would be overkill to add testing for every generic parsing error's detailed mode at this point.
But anyways, this means I'm not sure if these give us "weird outputs". However, I think they should be ok to leave with code printing, since there isn't good reason to believe all these `ParserContext` methods are given bad/meaningless `YYLTYPE` locations. I went ahead and manually tested some that I was concerned might print large blocks of code, but they look fine and printed single lines.

Reply:
Additionally, I'm adding code printing to `BisonSyntaxError` which seems to work well along the same lines as `ErrorParsing`.